### PR TITLE
Direct deploy information styling

### PR DESCRIPTION
**Pull request by Lukewh**
Supercedes https://github.com/juju/juju-gui/pull/2955

### CODE REVIEW

**Comment by Lukewh** [comment]
## QA

- Go to /new?dd=bundle/realtime-syslog-analytics there should be an image on the left and a title and description on the right.
- Got to /new?dd=nagios there should be a title and description.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7753/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7754/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7755/

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js [hunk 1]
```diff
@@ -157,7 +157,11 @@ YUI.add('deployment-flow', function() {
       const groupedChanges = this.props.groupedChanges;
       switch (section) {
         case 'model-name':
-          completed = false;
+          completed = this.props.acl.isReadOnly() ||
+            (
+              (this.state.modelName !== 'mymodel' || hasCloud)
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: Ah... Basically if someone doesn't change the default name of 'mymodel' we assume that adding a cloud implicitly indicates they're done and happy with the name (they can obviously change it afterwards, but for 'green tick progress' that section is "done").

File: jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js [hunk 2]
```diff
@@ -39,48 +39,80 @@ YUI.add('deployment-direct-deploy', function() {
...
+
+        let url;
+        let link;
+        try {
+          url = window.jujulib.URL.fromLegacyString(ddEntityId);
+        } catch(_) {
+          url = window.jujulib.URL.fromString(ddEntityId);
         }
+        if (url) {
+          link = (<a href={`/${url.path()}`}
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js]: do we need to use `state.generatePath` or whatever it is here to generate this URL correctly if the gui is not deployed under "/"?

Comment (Lukewh) [inline jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js]: @hatched ^

Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js]: Good catch @huwshimi , yes @Lukewh we'll need to generate a proper url here as in gijoe the root path isn't /

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7789/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7790/

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**